cms/djangoapps/models/settings/course_metadata.py

```diff
@@ -282,6 +286,37 @@ def validate_and_update_from_json(cls, block, jsondict, user, filter_tabs=True):
 
         return did_validate, errors, updated_data
 
+    @classmethod
+    def fill_teams_user_partitions_ids(cls, course_key, settings_dict):
+        """
+        Fill the `dynamic_user_partition_id` in the team settings if it is not set.
+
+        This is used by the Dynamic Team Partition Generator to create the dynamic user partitions
+        based on the team-sets defined in the course.
+        """
+        if not CONTENT_GROUPS_FOR_TEAMS.is_enabled(course_key):
+            return
+
+        teams_configuration_model = settings_dict.get('teams_configuration', {})
+        if teams_configuration_model == {}:
+            return
+        json_value = teams_configuration_model.get('value')
+        if json_value == '':
+            return
+
+        proposed_topics = json_value.get('topics')
+
+        if proposed_topics is None:
+            proposed_teamsets = json_value.get('team_sets')
+            if proposed_teamsets is None:
+                return
+            else:
+                proposed_topics = proposed_teamsets
```

Comment:
Note to reviewer: I copied how team_sets were loaded, so we could encapsulate this into its own function.